### PR TITLE
ngMessages leak

### PR DESCRIPTION
**Pull request by petebacondarwin**
Simple Plunker to demo: https://plnkr.co/edit/Ptn5iqTFcpxdu2sGPtFC?p=preview

### CODE REVIEW

File: src/ngMessages/messages.js [hunk 1]
```diff
@@ -719,6 +715,10 @@ function ngMessageDirectiveFactory() {
             }
           }
         });
+
+        scope.$on('$destroy', function() {
```
Comment (Narretz) [inline src/ngMessages/messages.js]: There should be a comment here why this extra listener is necessary. 
Btw, is this the new scope that belongs only to the message? Or the parent scope?

File: test/ngMessages/messagesSpec.js [hunk 2]
```diff
@@ -636,6 +636,27 @@ describe('ngMessages', function() {
...
   );
 
+  it('should unregister the ngMessage even if it was never attached',
+    inject(function($compile, $rootScope) {
+      var html =
+        '<div ng-messages="items">' +
+          '<div ng-if="show"><div ng-message="x">ERROR</div></div>' +
+        '</div>';
+
+      element = $compile(html)($rootScope);
```
Comment (Narretz) [inline test/ngMessages/messagesSpec.js]: There could be an expectation here that checks that the messge isn't initially attached / in the DOM

**Review by Narretz — CHANGES_REQUESTED** [review]
Looks good! Always nice when a fix doesn't involve adding more code.

File: src/ngMessages/messages.js (hunk 1 above)
Comment (petebacondarwin) [reply src/ngMessages/messages.js]: I'll add a comment explaining this.

This directive (and the `ngMessages` directive) do not create new scopes. So the scope we are listening to is the one that contains this `ngMessage` directive. E.g. the scope created by an `ngIf` or `ngRepeat` directive that wraps this `ngMessage`.

File: test/ngMessages/messagesSpec.js (hunk 2 above)
Comment (petebacondarwin) [reply test/ngMessages/messagesSpec.js]: Added

**Review by Narretz — APPROVED** [review]

**Comment by Narretz** [comment]
LGTM. I will merge once Travis has recovered - the builds are currently delayed because of an issue at their end

**Pull request merged**